src/Manager.hpp

```diff
@@ -219,14 +227,29 @@ inline std::vector<Entity*> ECS::EntitiesWith(Ts&& ...types)
 {
 	// build bitfield flags for this search
 	std::vector<std::string> componentNames;
-	componentNames = this->GetComponentNames(componentNames, std::forward<Ts>(types)...);
+	this->GetComponentNames(&componentNames, std::forward<Ts>(types)...);
+	
+	if (componentNames.size() == 0)
+	{
+		// Asking for all entities
+		std::vector<Entity*> requestedEntities;
+		for (auto e : this->entities)
+		{
+			requestedEntities.push_back(&e);
+		}
+		return requestedEntities;
```

Comment:
Minor thing on this block above, but given that we're just sending back the entire list of entities, we can probably just do `return this->entities`

Reply:
return turn is of type `std::vector<Entity*>`

`entities` is of type `std::vector<Entity>` - is there a more obvious way to convert it the expected (pointer) type?

Reply:
Return types are not the same, resolving comment per conversation.